ui/raidboss/data/06-ew/trial/asura.ts

```diff
@@ -1,14 +1,197 @@
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Asura Ephemerality',
+      type: 'Ability',
+      netRegex: { id: '8C96', source: 'Asura', capture: false },
+      suppressSeconds: 2,
+      alertText: (_data, _matches, output) => output.avoidClones!(),
+      outputStrings: {
+        'avoidClones': {
+          en: 'Avoid clones',
+        },
+      },
+    },
+    {
+      id: 'Asura Pedestal Purge',
+      type: 'StartsUsing',
+      netRegex: { id: '8C81', source: 'Asura', capture: false },
+      response: Responses.getOut(),
+    },
+    {
+      id: 'Asura Wheel of Deincarnation',
+      type: 'StartsUsing',
+      netRegex: { id: '8C83', source: 'Asura', capture: false },
+      response: Responses.getIn(),
+    },
+    // TODO: Possibly fire the Iconic Execution alert sooner using the tether
+    // or 8CB2 (Iconic Execution image jump to new cardinal). But it's not much of a gain,
+    // especially for the jumps.
+    {
+      id: 'Asura Iconic Execution',
+      type: 'StartsUsing',
+      netRegex: { id: Object.keys(imageIconographyIds), source: 'Asura Image', capture: true },
+      alertText: (_data, matches, output) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+        const imageLoc = Directions.xyToCardinalDirOutput(x, y, centerX, centerY);
+        const iconType = imageIconographyIds[matches.id];
+
+        if (iconType === undefined)
+          return;
+
+        const spotOutput = output[iconType]!();
+        let dirsOutput: string;
+
+        if (iconType === 'in')
+          dirsOutput = output[imageLoc]!();
+        else if (iconType === 'out')
+          dirsOutput = output[outSafeSpots[imageLoc]]!();
+        else { // sides
+          const [dir1, dir2] = sidesSafeSpots[imageLoc];
+          const dir1Output = output[dir1 ?? 'unknown']!();
+          const dir2Output = output[dir2 ?? 'unknown']!();
+          dirsOutput = output.doubledirs!({ dir1: dir1Output, dir2: dir2Output });
+        }
+
+        return output.text!({ dirs: dirsOutput, spot: spotOutput });
+      },
+      outputStrings: {
+        text: {
+          en: 'Go ${dirs} ${spot}',
+        },
+        doubledirs: {
+          en: '${dir1} / ${dir2}',
+        },
+        in: {
+          en: '(under image)',
+        },
+        out: {
+          en: '(away from image)',
+        },
+        sides: {
+          en: '(sides of image)',
+        },
+        ...Directions.outputStringsCardinalDir,
+      },
+    },
+    {
+      id: 'Asura Face of Wrath',
+      type: 'StartsUsing',
+      netRegex: { id: '8C90', source: 'Asura', capture: false },
+      alertText: (_data, _matches, output) => output.wrath!(),
+      outputStrings: {
+        'wrath': {
+          en: 'Stand in blue half',
+        },
+      },
+    },
+    {
+      id: 'Asura Face of Delight',
+      type: 'StartsUsing',
+      netRegex: { id: ['8C93', '8C95'], source: 'Asura', capture: false },
```

Comment:
Did you happen to find out whether the two IDs for this were indeed whether the axis between the halves is N/S vs E/W? I forgot to check through my recordings.

Reply:
Over the 8 or so runs, here's the ids I've seen.  
For Face of Wrath (red face, blue safe):
`8C90` is Red E, Blue W (the first Many Faces cast is always 8C90)
`8C92` = Red N, Blue S (can happen late in the encounter/once looping)

For Face of Delight (blue face, red safe):
`8C93` is Red N, Blue S
`8C95` = Blue N, Red S